api/src/pcapi/core/search/__init__.py

```diff
@@ -381,6 +385,42 @@ def index_offers_of_venues_in_queue() -> None:
         logger.exception("Could not index offers of venues from queue")
 
 
+def get_base_query_for_offer_indexation() -> BaseQuery:
+    return (
+        offers_models.Offer.query.options(
+            sa.orm.joinedload(offers_models.Offer.venue).joinedload(offerers_models.Venue.managingOfferer)
+        )
+        .options(sa.orm.joinedload(offers_models.Offer.criteria))
+        .options(sa.orm.joinedload(offers_models.Offer.mediations))
+        .options(sa.orm.joinedload(offers_models.Offer.product))
+        .options(sa.orm.joinedload(offers_models.Offer.stocks))
+    )
+
+
+def get_base_query_for_last_30_days_bookings() -> BaseQuery:
+    return (
+        sa.select(
+            offers_models.Offer.id.label("offer_id"), sa.func.count(bookings_models.Booking.id).label("bookings_number")
+        )
+        .select_from(offers_models.Offer)
+        .join(offers_models.Stock, offers_models.Stock.offerId == offers_models.Offer.id)
+        .join(bookings_models.Booking, bookings_models.Booking.stockId == offers_models.Stock.id)
```

Comment:
(détail / question de goût) En plus court et qui me semble plus simple à lire : 

```suggestion
        .join(offers_models.Offer.stocks)
        .join(offers_models.Stock.bookings)
```